urlac47192b66_ggrandes_bouncer_tgz-pJ8-org_javastack_bouncer_ClusterTestJ8/src/org/javastack/bouncer/jmx/AnnotatedStandardMBean.java

```diff
@@ -0,0 +1,155 @@
+	}
+
+	/** Instance where the MBean interface is implemented by another object. */
+	public <T> AnnotatedStandardMBean(T impl, Class<T> mbeanInterface) throws NotCompliantMBeanException {
+		super(impl, mbeanInterface);
+	}
+
+	/** Instance where the MBean interface is implemented by this object. */
+	protected AnnotatedStandardMBean(Class<?> mbeanInterface) throws NotCompliantMBeanException {
+		super(mbeanInterface);
+	}
+
+	/** {@inheritDoc} */
+	@Override
+	protected String getDescription(MBeanAttributeInfo info) {
+
+		String descr = info.getDescription();
+		Method m = getMethod(getMBeanInterface(), "get" + info.getName().substring(0, 1).toUpperCase()
+				+ info.getName().substring(1));
+		if (m == null)
+			m = getMethod(getMBeanInterface(), "is" + info.getName().substring(0, 1).toUpperCase()
+					+ info.getName().substring(1));
+		if (m == null)
+			m = getMethod(getMBeanInterface(), "does" + info.getName().substring(0, 1).toUpperCase()
+					+ info.getName().substring(1));
+
+		if (m != null) {
+			Description d = m.getAnnotation(Description.class);
+			if (d != null)
+				descr = d.value();
+		}
+		return descr;
+	}
+
+	/** {@inheritDoc} */
+	@Override
+	protected String getDescription(MBeanOperationInfo op) {
+
+		String descr = op.getDescription();
+		Method m = getMethod(op);
+		if (m != null) {
+			Description d = m.getAnnotation(Description.class);
+			if (d != null)
+				descr = d.value();
+		}
+		return descr;
+	}
+
+	/** {@inheritDoc} */
+	@Override
+	protected String getParameterName(MBeanOperationInfo op, MBeanParameterInfo param, int paramNo) {
+		String name = param.getName();
+		Method m = getMethod(op);
+		if (m != null) {
+			for (Annotation a : m.getParameterAnnotations()[paramNo]) {
+				if (Description.class.isInstance(a))
+					name = Description.class.cast(a).value();
+			}
+		}
+		return name;
+	}
+
+	/**
+	 * Extracts the Method from the MBeanOperationInfo
+	 * 
+	 * @param op
+	 * @return
+	 */
+	private Method getMethod(MBeanOperationInfo op) {
+		final MBeanParameterInfo[] params = op.getSignature();
+		final String[] paramTypes = new String[params.length];
+		for (int i = 0; i < params.length; i++)
+			paramTypes[i] = params[i].getType();
+
+		return getMethod(getMBeanInterface(), op.getName(), paramTypes);
+	}
+
+	/**
+	 * Returns the Method with the specified name and parameter types for the
+	 * given class, null if it doesn't exist.
+	 * 
+	 * @param mbean
+	 * @param method
+	 * @param params
+	 * @return
+	 */
+	private static Method getMethod(Class<?> mbean, String method, String... params) {
+		try {
+			final ClassLoader loader = mbean.getClassLoader();
+			final Class<?>[] paramClasses = new Class<?>[params.length];
+			for (int i = 0; i < params.length; i++) {
+				paramClasses[i] = primitives.get(params[i]);
+				if (paramClasses[i] == null)
+					paramClasses[i] = Class.forName(params[i], false, loader);
+			}
+			return mbean.getMethod(method, paramClasses);
+		} catch (RuntimeException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)